osquery/tables/system/windows/disk_info.cpp

```diff
@@ -22,9 +22,10 @@ QueryData genDiskInfo(QueryContext& context) {
   Row r;
   QueryData results;
 
-  const WmiRequest wmiSystemReq("select * from Win32_DiskDrive");
-  const std::vector<WmiResultItem>& wmiResults = wmiSystemReq.results();
-  if (wmiResults.empty()) {
+  const Expected<WmiRequest, WmiError> wmiSystemReq =
+      WmiRequest::CreateWmiRequest("select * from Win32_DiskDrive");
+  const std::vector<WmiResultItem>& wmiResults = wmiSystemReq->results();
```

Comment:
Should the dereference of `wmiSystemReq` wait until the check below for `if (!wmiSystemReq` has happened?

Reply:
(If so, there are similar sequences in other files in this PR)

Reply:
Fixed all instances I believe. I had realized that halfway through but had neglected to go back and make it consistent.